File: 2_day/GCD/GCD/ViewController.m
//
//  ViewController.m
//  GCD
//
//  Created by 张正雨 on 2017/5/21.
//  Copyright © 2017年 张正雨. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    [self concurrentSync];
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event{
    //创建并发队列
    dispatch_queue_t queue = dispatch_queue_create("muQueue", DISPATCH_QUEUE_CONCURRENT);
    //在队列里面添加任务
    //同步任务
    dispatch_sync(queue, ^{
        for (int i = 0 ; i < 5; i++) {
            NSLog(@"---1---%@",[NSThread currentThread]);
        }
    });
    dispatch_sync(queue, ^{
        for (int i = 0 ; i < 5; i++) {
            NSLog(@"---2---%@",[NSThread currentThread]);
        }
    });
    dispatch_sync(queue, ^{
        for (int i = 0 ; i < 5; i++) {
            NSLog(@"---3---%@",[NSThread currentThread]);
        }
    });
}
/**
 并发队列 + 同步任务：没有开启新的线程，任务是逐个执行的
 */
- (void)concurrentSync{
    //创建并发队列
    dispatch_queue_t queue = dispatch_queue_create("muQueue", DISPATCH_QUEUE_CONCURRENT);
    //在队列里面添加任务
    //同步任务
    dispatch_sync(queue, ^{
        for (int i = 0 ; i < 5; i++) {
            NSLog(@"---1---%@",[NSThread currentThread]);
        }
    });
    dispatch_sync(queue, ^{
        for (int i = 0 ; i < 5; i++) {
            NSLog(@"---2---%@",[NSThread currentThread]);
        }
    });
    dispatch_sync(queue, ^{
        for (int i = 0 ; i < 5; i++) {
            NSLog(@"---3---%@",[NSThread currentThread]);
        }
    });
}


@end

File: 1_day/010-字典/010-字典/ViewController.swift
//
//  ViewController.swift
//  010-字典
//
//  Created by 张正雨 on 2017/5/18.
//  Copyright © 2017年 张正雨. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        demo1()
        demo2()
        demo3()
        demo4()
    }
    //合并
    func demo1() -> () {
        var dict:[String:Any] = ["name":"张","age":5,"title":"jajaja"]
        let dict2 = ["avater":"werfgf"]
        for (key,value) in dict2 {
            dict[key] = value
        }
        print(dict)
        
    }
    //遍历
    func demo2() -> () {
        var dict:[String:Any] = ["name":"张","age":5,"title":"jajaja"]
        for e in dict {
            print(e)
        }
    }
    //增删改
    func demo3() -> () {
        var dict:[String:Any] = ["name":"张","age":5]
        
        dict["title"] = "dage"
        print(dict)
        
        dict["name"] = "😁"
        print(dict)
        
        dict.removeValue(forKey: "age")
        print(dict)
    }
    //定义
    func demo4() -> () {
        let dict:[String:Any] = ["name":"张","age":5]
        print(dict)
        
        let array:[[String:Any]] = [["name":"张","age":5],
                                    ["name":"王","age":33]]
        print(array)
        print(array)
    }
}

